src/main/java/uk/gov/ons/census/casesvc/service/UacProcessor.java

```diff
@@ -57,24 +57,27 @@ public UacQidLink saveUacQidLink(Case caze, int questionnaireType) {
     return uacQidLink;
   }
 
-  public void logEvent(UacQidLink uacQidLink, String eventDescription) {
+  public void logEvent(
+      UacQidLink uacQidLink, String eventDescription, LocalDateTime eventMetaDataDateTime) {
     uk.gov.ons.census.casesvc.model.entity.Event loggedEvent =
         new uk.gov.ons.census.casesvc.model.entity.Event();
     loggedEvent.setId(UUID.randomUUID());
-    loggedEvent.setEventDate(new Date());
+    loggedEvent.setEventDate(eventMetaDataDateTime);
+    loggedEvent.setRmEventProcessed(LocalDateTime.now());
     loggedEvent.setEventDescription(eventDescription);
     loggedEvent.setUacQidLink(uacQidLink);
     eventRepository.save(loggedEvent);
   }
 
-  public void emitUacUpdatedEvent(UacQidLink uacQidLink, Case caze) {
+  public void emitUacUpdatedEvent(UacQidLink uacQidLink, Case caze, Boolean active) {
```

Comment:
1. I hate the use of `Boolean` except where were' planning on passing a null as one of three options

2. Can't we overload this method so that we keep the original 2-argument version, calling the second one with null or whatever default we want?